sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,217 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division
+
+import numpy as np
+from ..externals.six.moves import xrange
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted, column_or_1d
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int (default=2)
```

Comment:
We wouldn't use `np.histogram` anyway: we don't care for the sums in each bin. But the criteria are straightforward to implement anyway.

I don't mind deferring this feature for future PRs, but I think we should consider the case of different numbers of bins per feature, as it's easily conceivable, and affects data structures.